src/licensedcode/detection.py

```diff
@@ -263,6 +370,278 @@ def matched_text(
+    for license_key in match.rule.license_keys():
+        lic = licenses.get(license_key)
+        result = {}
+        detected_licenses.append(result)
+        result['key'] = lic.key
+        result['score'] = match.score()
+        result['name'] = lic.name
+        result['short_name'] = lic.short_name
+        result['category'] = lic.category
+        result['is_exception'] = lic.is_exception
+        result['is_unknown'] = lic.is_unknown
+        result['owner'] = lic.owner
+        result['homepage_url'] = lic.homepage_url
+        result['text_url'] = lic.text_urls[0] if lic.text_urls else ''
+        result['reference_url'] = license_url_template.format(lic.key)
+        result['scancode_text_url'] = SCANCODE_LICENSE_TEXT_URL.format(lic.key)
+        result['scancode_data_url'] = SCANCODE_LICENSE_DATA_URL.format(lic.key)
+
+        spdx_key = lic.spdx_license_key
+        result['spdx_license_key'] = spdx_key
+
+        if spdx_key:
+            is_license_ref = spdx_key.lower().startswith('licenseref-')
+            if is_license_ref:
+                spdx_url = SCANCODE_LICENSE_TEXT_URL.format(lic.key)
+            else:
+                spdx_key = lic.spdx_license_key.rstrip('+')
+                spdx_url = SPDX_LICENSE_URL.format(spdx_key)
+        else:
+            spdx_url = ''
+        result['spdx_url'] = spdx_url
+        result['start_line'] = match.start_line
+        result['end_line'] = match.end_line
+        matched_rule = result['matched_rule'] = {}
+        matched_rule['identifier'] = match.rule.identifier
+        matched_rule['license_expression'] = match.rule.license_expression
+        matched_rule['licenses'] = match.rule.license_keys()
+        matched_rule['referenced_filenames'] = match.rule.referenced_filenames
+        matched_rule['is_license_text'] = match.rule.is_license_text
+        matched_rule['is_license_notice'] = match.rule.is_license_notice
+        matched_rule['is_license_reference'] = match.rule.is_license_reference
+        matched_rule['is_license_tag'] = match.rule.is_license_tag
+        matched_rule['is_license_intro'] = match.rule.is_license_intro
+        matched_rule['has_unknown'] = match.rule.has_unknown
+        matched_rule['matcher'] = match.matcher
+        matched_rule['rule_length'] = match.rule.length
+        matched_rule['matched_length'] = match.len()
+        matched_rule['match_coverage'] = match.coverage()
+        matched_rule['rule_relevance'] = match.rule.relevance
+        # FIXME: for sanity this should always be included?????
+        if include_text:
+            result['matched_text'] = matched_text
+    return detected_licenses
+
+
+def matches_compact_repr(matches):
+    """
+    Return a string representing a list of license matches in a compact way.
+    """
+    return ', '.join(m.rule.identifier for m in matches)
+
+
+def is_correct_detection(license_matches):
+    """
+    Return True if all the license matches in a file-region are correct
```

Comment:
Do not expose too much of the internal implementation in your docstring,
```suggestion
    Return True if all the matches in ` license_matches` are correct e.g., highly accurate and not ambiguous.
```

Reply:
Yes.